exec/cnex/nstring.c

```diff
@@ -221,6 +233,37 @@ BOOL string_startsWith(TString *self, TString *s)
     return TRUE;
 }
 
+BOOL string_endsWithNoCase(TString *self, TString *s)
+{
+    size_t i, si;
+    if (self->length < s->length) {
+        return FALSE;  // Can't possibly END with s, if self isn't even large enough to contain s.
+    }
+
+    for (si = 0, i = self->length - s->length; i < self->length; si++, i++) {
+        if (tolower(self->data[i]) != tolower(s->data[si])) {
```

Comment:
Let me tell you a little story about this. In the Turkish language, `ı` and `i` are different letters. Similarly, `I` and `İ` are their upper case equivalents. See https://en.wikipedia.org/wiki/Dotted_and_dotless_I for more information. So if your OS locale is set to English, then "file" and "FILE" compare the same using this comparison algorithm. But if your OS locale is set to Turkish, then "file" and "FILE" are different. (In that case, "file" and "FİLE" would compare the same). There isn't anything you need to do at this point, it's just being aware.